src/core/i18n/utils.js

```diff
@@ -173,3 +175,26 @@ export function getLanguage({ lang, acceptLanguage } = {}) {
   // - normalization e.g: en-us -> en-US.
   return { lang: sanitizeLanguage(userLang), isLangFromHeader };
 }
+
+// moment uses locales like "en-gb" whereas we use "en_GB".
+export function makeMomentLocale(locale) {
+  return locale.replace('_', '-').toLowerCase();
+}
+
+// Create an i18n object with a translated moment object available we can
+// use for translated dates across the app.
+export function makeI18n(i18nData, _Jed = Jed) {
+  const i18n = new _Jed(i18nData);
+
+  // This adds the correct moment locale for the active locale so we can get
+  // localised dates, times, etc.
+  if (i18n.options && typeof i18n.options._momentDefineLocale === 'function') {
+    i18n.options._momentDefineLocale();
+    moment.locale(makeMomentLocale(i18n.options.locale_data.messages[''].lang));
```

Comment:
It looks like you can actually set a locale for an [instance of moment](http://momentjs.com/docs/#/i18n/instance-locale/). Maybe that would be better?

```js
const i18n = new _Jed(i18nData);
i18n.moment = moment();
if (i18n.options && typeof i18n.options._momentDefineLocale === 'function') {
  i18n.options._momentDefineLocale();
  i18n.moment.locale(makeMomentLocale(i18n.options.locale_data.messages[''].lang));
}
return i18n;